deterministic-build-wrappers/rustc

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# rustc calculates a unique metadata tag to mangle symbols which includes the
+# actual path to the crate. This breaks our deterministic builds as we depend
+# on a copy of rust-lightning via a path. We insert this shim between cargo and
+# rustc and edit the metadata tag for rust-lightning.
+# While we could just set RUSTFLAGS="-C metadata=42", this would break if we
+# ever (indirectly) depend on multiple versions of the same crate.
```

Comment:
What does "indirectly" mean here? Does this sentence refer to the situation where e.g. `lightning-net-tokio` depends on tokio 1.0 but `lightning-block-sync` depends on tokio 1.1?

Reply:
Right, or that rust-bitcoin depends on something which depends on a different version. It doesn't have to specifically be directly a dependency of rust-lightning or LDK stuff.